sympy/logic/boolalg.py

```diff
@@ -2822,20 +2852,22 @@ def simplify_logic(expr, form=None, deep=True, force=False):
     if form not in (None, 'cnf', 'dnf'):
         raise ValueError("form can be cnf or dnf only")
     expr = sympify(expr)
-    # check for quick exit: right form and all args are
+    # check for quick exit if form is given: right form and all args are
     # literal and do not involve Not
-    isc = is_cnf(expr)
-    isd = is_dnf(expr)
-    form_ok = (
-        isc and form == 'cnf' or
-        isd and form == 'dnf')
-    if form_ok and all(is_literal(a)
-            for a in expr.args):
-        return expr
+    if form:
+        form_ok = False
+        if form == 'cnf':
+            form_ok = is_cnf(expr)
+        elif form == 'dnf':
+            form_ok = is_dnf(expr)
+
+        if form_ok and all(is_literal(a)
+                for a in expr.args):
+            return expr
     if deep:
         variables = _find_predicates(expr)
         from sympy.simplify.simplify import simplify
-        s = [simplify(v) for v in variables]
+        s = tuple(map(simplify, variables))
```

Comment:
Generally I prefer list comprehensions for things like this just because it looks clearer to me. Is there actually a speed advantage here?

Does the change from list to tuple have any observable effect on the output from this function?

In my experience `map` is faster if the iterable and the function are all builtins but the speed difference evaporates once the function is a python function or if the iterable is something like a generator.

Reply:
I've bench-marked some things, not all, and tried to generalize it and it seems like tuples are faster, even with your own functions (as long as it is not lambdas, `map` should not be slower from what I've read). In this file it was a mix of `map` and list comprehensions for exactly this case, so I thought I'd unify it and went for this.